lang/python/gunicorn/Makefile

```diff
@@ -10,43 +10,115 @@ include $(TOPDIR)/rules.mk
 PKG_NAME:=gunicorn
 PKG_VERSION:=19.9.0
 PKG_RELEASE=2
-PKG_LICENSE:=MIT
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/g/gunicorn
 PKG_HASH:=fa2662097c66f920f53f70621c6c58ca4a3c4d3434205e608e121b5b3b71f4f3
+
 PKG_MAINTAINER:=Alexandru Ardelean <ardeleanalex@gmail.com>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(BUILD_VARIANT)-$(PKG_NAME)-$(PKG_VERSION)
+
+PYTHON_DO_NOT_COMPILE:=1
```

Comment:
This will cause target Python to compile the code on each run, which isn't great for performance.

Is there any way we can work around or ignore the bytecode compilation error?

Reply:
yes; but this is how things were happening before this change for the Python2 version of gunicorn;
so, nothing is worse with regard to this change;

and i also had some internal thoughts about how-to-do things;
and i am glad that i can share some of them; makes me feel less insane/undecided

there are several other options:
1. compile everything except for `/usr/lib/python2.7/site-packages/gunicorn/workers/_gaiohttp.py`
2. patch `_gaiohttp.py` so that it compiles for Python2 as well; may not be great for Python3 as it may depend on that behavior
3. drop `_gaiohttp.py` from the package completely, until we move this to Python3

there may be other options here, but I can't think of them atm;

but the idea is that:
* Python2 should be going away
* the change should not make things worse for gunicorn (than they were before)
* i suspect people use gunicorn as a standalone-external sever (so gunicorn3 would be faster since it's byte-compiled)
* there are 2 users of gunicorn atm: `asu` and the seafile suite;  admit there may be users of gunicorn just using it for something else; but since it was ok for them without byte-compiled stuff, gunicorn3 should be better for them as well;

Reply:
I think Debian drops the file from the Python 2 package. That’s probably the easiest way to handle it.

I believe `PyPackage/python-gunicorn/filespec` is processed before the bytecode compilation step, so removing `_gaiohttp.py` should be relatively easy.

I think we can also keep the “do not compile” functionality, in case another package needs it.

Reply:
> I think Debian drops the file from the Python 2 package. That’s probably the easiest way to handle it.

hmmm; good idea;
i'll do that then


> I think we can also keep the “do not compile” functionality, in case another package needs it.

hmm; i wouldn't leave dead-code in there;
i would drop it

thanks for the ideas